onnxscript/rewriter/pattern.py

```diff
@@ -800,63 +678,68 @@ def __init__(self, function: Callable) -> None:
     def function(self) -> Callable:
         return self._function
 
-    def get_pattern(self, *variables: Sequence[Var]) -> tuple[NodePattern, int]:
+    def get_pattern(self, variables: Sequence[Var]) -> tuple[NodePattern, int]:
         node_output_pattern = self._function(*variables)
         return _handle_pattern_return_value(node_output_pattern)
 
 
+class RewriterContext:
+    """Context parameter used to build the replacement pattern."""
+
+    # TODO(justinchuby): Merge with the rest of pattern building methods
+    def __init__(self):
+        self.tape = _tape.Tape()
+
+    def __getattr__(self, op_type: str) -> Any:
+        return lambda *args, **kwargs: self._make_node(op_type, args, kwargs)
+
+    def _make_node(self, op_type: str, inputs: Sequence[ir.Value], kwargs: dict[str, Any]):
+        domain = kwargs.pop("domain", "")
```

Comment:
Would versioning be a problem when it comes to custom domain? Now we don't have version info to add into model.opset_imports?

Reply:
yes, we also need to support version as a parameter. That's pending on my list, will handle it.